src/lr_scheduler/cosine_annealing_lr.rs

```diff
@@ -0,0 +1,51 @@
+use super::Schedule;
+
+pub struct CosineAnnealingLR {
+    pub base_lr: f64,
+    pub t_max: usize,
+    pub eta_min: f64,
+}
+
+impl Default for CosineAnnealingLR {
+    fn default() -> Self {
+        CosineAnnealingLR {
+            base_lr: 0.05,
+            t_max: 5,
+            eta_min: 0.0,
+        }
+    }
+}
+
+impl Schedule for CosineAnnealingLR {
+    fn get_lr(&self, lr: f64, last_epoch: usize) -> f64 {
+        if last_epoch == 0 {
+            return lr;
+        }
+
+        let pi = std::f64::consts::PI;
+        let lr_range = self.base_lr - self.eta_min;
+        let wavelen_percent = pi * (last_epoch as f64) / (self.t_max as f64);
+
+        if last_epoch == 1 {
+            self.eta_min + lr_range * (1.0 + wavelen_percent.cos()) / 2.0
+        } else if (last_epoch as i32 - self.t_max as i32 - 1) % (2 * self.t_max as i32) == 0 {
```

Comment:
@LazaroHurtado Couldn't decide how to deal with this expression so I left it as is for this first commit.

Reply:
looks like you are type casting `t_max` twice and in other conditions, might be best to do this outside the `else if`